src/SchemaReader.php

```diff
@@ -1170,6 +1173,32 @@ private function loadImport(
         return $this->loadImportFresh($namespace, $schema, $file);
     }
 
+    private function loadRedefine(Schema $schema, DOMElement $node): Closure
+    {
+        $base = urldecode($node->ownerDocument->documentURI);
+        $file = UrlUtils::resolveRelativeUrl($base, $node->getAttribute('schemaLocation'));
```

Comment:
are you sure that `redefine` semantics is applied on schemaLocation` and on the target namespace?

Reply:
I am not sure that it implements all semantics defined for it but it works for the case where additional Elements and Types are added to a namespace. this is what is covered in the test cases from the original PR and also in the case I extracted from the spec I am currently working with.